### PR TITLE
Added torchaudio.datasets.SPEECHCOMMANDS()

### PR DESCRIPTION
**Pull request by kaiidams**
- Added torchaudio.datasets.SPEECHCOMMANDS()
- Made some of internal functions of torchaudio.datasets.YESNO() sharable.

### CODE REVIEW

**Comment by NiklasGustafsson** [comment]
@kaiidams -- while you're at it, could you also take a look at #637

**Review by NiklasGustafsson — APPROVED** [review]

File: src/TorchSharp/TorchAudio/Datasets/SpeechCommandsDatasetItem.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using System.Linq;
+using System.Text;
+
+namespace TorchSharp
+{
+    public static partial class torchaudio
+    {
+        public static partial class datasets
+        {
+            public class SpeechCommandsDatasetItem
```
Comment (NiklasGustafsson) [inline src/TorchSharp/TorchAudio/Datasets/SpeechCommandsDatasetItem.cs:14]: Doc comments?

File: src/TorchSharp/TorchAudio/Datasets/Utils.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace TorchSharp
+{
+    public static partial class torchaudio
+    {
+        public static partial class datasets
+        {
+            public static partial class utils
+            {
+                internal static void extract_archive(string from_path, string to_path)
```
Comment (NiklasGustafsson) [inline src/TorchSharp/TorchAudio/Datasets/Utils.cs:18]: Maybe this should be a public method? And add doc comments.

**Review by NiklasGustafsson — CHANGES_REQUESTED** [review]

File: src/TorchSharp/TorchAudio/Datasets/SpeechCommandsDatasetItem.cs (hunk 1 above)
Comment (kaiidams) [reply src/TorchSharp/TorchAudio/Datasets/SpeechCommandsDatasetItem.cs:14]: Added doc comments

File: src/TorchSharp/TorchAudio/Datasets/Utils.cs (hunk 2 above)
Comment (kaiidams) [reply src/TorchSharp/TorchAudio/Datasets/Utils.cs:18]: Added doc comment. `torchtext.utils.extract_archive()` is a documented API of PyTorch https://pytorch.org/text/stable/_modules/torchtext/utils.html#extract_archive
but `torchaudio.datasets.utils.extract_archive()` is not documented.

**Review by NiklasGustafsson — APPROVED** [review]

**Pull request merged**